src/cmd-prune-containers

```diff
@@ -0,0 +1,171 @@
+            days = value
+        case _:
+            raise ValueError(f"Invalid unit '{match.group(2)}'. Please use y (years), m (months), w (weeks), or d (days).")
+
+    return days
+
+
+def skopeo_delete(repo, image, auth):
+
+    skopeo_args = ["skopeo", "delete", f"docker://{repo}:{image}"]
+    if auth is not None:
+        skopeo_args.append(f"--authfile {auth}")
+
+    subprocess.run(skopeo_args)
+
+
+# FIXME : move to cosa_lib
+# https://github.com/coreos/coreos-assembler/pull/3798/files#r1673481990
+
+def parse_fcos_version(version):
+    m = regexp.match(r'^([0-9]{2})\.([0-9]{8})\.([0-9]+)\.([0-9]+)$', version)
+    if m is None:
+        raise ValueError(f"Incorrect versioning for FCOS build {version}")
+    try:
+        timestamp = datetime.datetime.strptime(m.group(2), '%Y%m%d')
+    except ValueError:
+        raise Exception(f"FCOS build {version} has incorrect date format. It should be in (%Y%m%d)")
+    return (timestamp, int(m.group(3)))
+
+
+def get_update_graph(stream):
+
+    url = f"https://builds.coreos.fedoraproject.org/updates/{stream}.json"
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise Exception(f"Could not download update graph for {stream}. HTTP {r.status_code}")
+    return r.json()
+
+
+class BarrierRelease(Exception):
+    pass
+
+
+def main():
+
+    args = parse_args()
+
+    # Load the policy file
+    with open(args.policy, "r") as f:
+        policy = yaml.safe_load(f)
+    if args.stream not in policy:
+        print(f"Stream {args.stream} is not defined in policy file.")
+        exit(1)
+    if 'containers' not in policy[args.stream]:
+        print(f"No containers section for {args.stream} stream in policy.")
+        exit(1)
+    policy = policy[args.stream]["containers"]
+
+    print(f"Pulling tags from {args.repository_url}")
+    # This is a JSON object:
+    # {"Repository": "quay.io/jbtrystramtestimages/fcos",
+    #    "Tags": [
+    #        "40.20"40.20240301.1.0",.....]}
+    tags_data = subprocess.check_output(["skopeo", "list-tags",
+                                        f"docker://{args.repository_url}"])
+
+    tags_json = json.loads(tags_data)
+    tags = tags_json['Tags']
+    # Compute the date before we should prune images
+    # today - prune-policy
+    today = datetime.datetime.now()
+    date_limit = today - relativedelta(days=convert_to_days(policy))
+    print(f"This will delete any images older than {date_limit} from the stream {args.stream}")
+
+    stream_id = STREAMS[args.stream]
+    barrier_releases = []
+    # Get the update graph for stable streams
+    if args.stream in ['stable', 'testing', 'next']:
+        update_graph = get_update_graph(args.stream)['releases']
+        # Keep only the barrier releases
+        # filter(lambda release: "barrier" in release["metadata"], update_graph)
+        for release in update_graph:
+            if "barrier" in release["metadata"]:
+                barrier_releases.append(release["version"])
```

Comment:
Minor/optional (related to above):

```python
barrier_releases = set([release["version"] for release in update_graph if "barrier" in release])
```

Reply:
To me it's the same, but harder to read :D  But i am not well versed in python.
I applied the suggestion

Reply:
That's fair. Wanted to show (what I think is) a more idiomatic way but also totally fine the more verbose way if it seems clearer to you.